### PR TITLE
Make `user-content-* ` consistent with github

### LINKED ISSUE #26367: anchors don't work when contains punctuation marks just like `（` or `(`
### Description

Markdown Heading ID contains Unicode `（` is inconsistent with Github.
For `#### test（1）` in gitea , the id is `"user-content-test-1"` and in github it is `"user-content-test1"`

The markdown below is available for jumping on github, but not for gitea.
```md
#### test（1）
to [test（1）](#test1)
```

#### gitea

`id = "user-content-test-1"`
8d42bfb9f7ea6f43e4f3b6b8f339aff

#### github

`id = "user-content-test1"`
89b1f42dcd9174ec189487dd55057a4

### Gitea Version

1.20.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

I was able to reproduce it using try.gitea.io.

### Database

None

### PR DESCRIPTION
**Pull request by CaiCandong**
Fix #26367
Related #19745

Thanks @lazyky for providing  test cases

### CODE REVIEW

File: modules/markup/common/footnote_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		{"tes￥& a", "tes-a"},
+		{"tes= a", "tes-a"},
+		{"tes|a", "tesa"},
+		{"tes\\a", "tesa"},
+		{"tes/a", "tesa"},
+	}
+	for _, test := range tests {
+		if got := CleanValue([]byte(test.param)); string(got) != test.expect {
+			t.Errorf("CleanValue(%q) = %q, want %q", test.param, got, test.expect)
+		}
```
Comment (wxiaoguang) [inline modules/markup/common/footnote_test.go]: It could be `assert.Equal(t, test.expect, CleanValue([]byte(test.param)), test.param)`

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
image

File: modules/markup/common/footnote_test.go (hunk 1 above)
Comment (CaiCandong) [reply modules/markup/common/footnote_test.go]: thanks

**Review by lunny — APPROVED** [review]

**Comment by CaiCandong** [comment]
> Merge or wait for [#26388 (comment)](https://github.com/go-gitea/gitea/pull/26388#issuecomment-1669081327)?

These three test cases have been added to the unit tests：[#64f5bb1ad9bbb690b7cb890a643611f413132d0d](https://github.com/go-gitea/gitea/pull/26388/commits/64f5bb1ad9bbb690b7cb890a643611f413132d0d)

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
> > Merge or wait for [#26388 (comment)](https://github.com/go-gitea/gitea/pull/26388#issuecomment-1669081327)?
> 
> These three test cases have been added to the unit tests：[#64f5bb1ad9bbb690b7cb890a643611f413132d0d](https://github.com/go-gitea/gitea/pull/26388/commits/64f5bb1ad9bbb690b7cb890a643611f413132d0d)，I apologize for not responding to comments in a timely manner.

Ah, I only saw the "fix lint" commit, didn't think it also added test cases :)

**Pull request merged**